File: solutions_1483485_0/Python/lubieparowki/speak.py
import os, sys

inFile = None
outFile = None

mapping = {}

def solveCase(caseNo):
	line = inFile.readline()
	translated = ""
	for x in line:
		if x.isalpha():
			translated += mapping[x]
		else:
			translated += x
	outFile.write("Case #{0}: {1}".format(caseNo, translated))

def main():
	t = int(inFile.readline().strip())
	for i in xrange(t):
		solveCase(i+1)

to   = "ejp mysljylc kd kxveddknmc re jsicpdrysirbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcdde kr kd eoya kw aej tysr re ujdr lkgc jv"
fr   = "our language is impossible to understandthere are twenty six factorial possibilitiesso it is okay if you want to just give up"

def prepareMapping():
	mapping['y'] = 'a'
	mapping['e'] = 'o'
	mapping['q'] = 'z'
	mapping['z'] = 'q'

	for x in range(len(to)):
		mapping[to[x]] = fr[x]

	#print len(mapping)

if __name__ == '__main__':
	inFile = open('in.txt','rt')
	outFile = open('out.txt', 'wt')
	prepareMapping()
	main()

File: solutions_1483485_0/C++/pinpin/speak.cpp
#include <vector>
#include <list>
#include <queue>
#include <map>
#include <set>
#include <string>
#include <iostream>
#include <sstream>
#include <cmath>
#include <algorithm>
#include <cstdio>
#include <cstdlib>

using namespace std;

int
main(void)
{
    int T;
    int i, j, n;
    char tbl[256];
    char buf[1024];

    cin >> T;
    string s0("ejp mysljylc kd kxveddknmc re jsicpdrysirbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcdde kr kd eoya kw aej tysr re ujdr lkgc jv");
    string s1("our language is impossible to understandthere are twenty six factorial possibilitiesso it is okay if you want to just give up");
    
    for(i=0;i<s0.size();i++) {
        tbl[s0[i]] = s1[i];
    }
    tbl['z'] = 'q';
    tbl['q'] = 'z';

    cin.getline(buf, sizeof(buf));
    for(i=1;i<=T;i++) {
        cin.getline(buf, sizeof(buf));
        for(j=0;j<1024;j++) {
            buf[j] = tbl[buf[j]];
        }
        // for(j=0;j<s.size();j++) {
        //     s[j] = tbl[s[j]];
        // }

        printf("Case #%d: %s\n", i, buf);
        // cout << "Case #" << i << ": " << s << endl;
    }
    
    return 0;
}
